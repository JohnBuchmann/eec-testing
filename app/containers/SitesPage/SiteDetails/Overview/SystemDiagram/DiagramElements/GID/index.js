/**
 * GID
 */

import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Colors } from 'Theme';
import { SiteStatusId } from 'Utils/enums/site';
import Status from '../Status';

/**
 * @property {Object} useStyles Stores the styles
 */
const useStyles = makeStyles({
  gidTitle: {
    fontFamily: 'Inter, sans-serif',
    color: Colors.surfCrest,
    fontSize: '10px',
    fontWeight: '300',
    lineHeight: 'normal',
    letterSpacing: '0px',
  },
});

/**
 * @method GID
 * Displays GID
 *
 * @param {number} x The X axis
 * @param {number} y The Y axis
 * @returns GID component
 */
function GID(props) {
  const classes = useStyles();
  const { x = 0, y = 0, isConnected } = props;

  const title = 'GID';
  const statusId = isConnected ? SiteStatusId.Ok : SiteStatusId.Faulted;
  const statusIconPositionX = x - 3;
  const statusIconPositionY = y + 7;
  const showIcon = !isConnected;

  return (
    <>
      <desc>GID</desc>
      <text x={x} y={y} className={classes.gidTitle}>
        {title}
      </text>
      {/* Status icon */}
      <Status
        x={statusIconPositionX}
        y={statusIconPositionY}
        showIcon={showIcon}
        width="24px"
        height="24px"
        statusId={statusId}
        roundness="3px"
      />
    </>
  );
}

GID.propTypes = {
  x: PropTypes.number,
  y: PropTypes.number,
  isConnected: PropTypes.bool,
};

export default GID;
