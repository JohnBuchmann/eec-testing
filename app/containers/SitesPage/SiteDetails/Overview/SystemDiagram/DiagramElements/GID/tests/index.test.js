import React from 'react';
import { shallow } from 'enzyme';
import GID from '..';

describe('GID', () => {
  it('should render a connected GID with default props', () => {
    const actual = shallow(<GID />);

    expect(actual).toMatchSnapshot();
  });

  it('should render a connected GID when isConnected param is true', () => {
    const mockPositionX = 100;
    const mockPositionY = 100;
    const isConnected = true;

    const actual = shallow(
      <GID x={mockPositionX} y={mockPositionY} isConnected={isConnected} />
    );

    expect(actual).toMatchSnapshot();
  });

  it('should render a connected GID when isConnected param is false', () => {
    const mockPositionX = 100;
    const mockPositionY = 100;
    const isConnected = false;

    const actual = shallow(
      <GID x={mockPositionX} y={mockPositionY} isConnected={isConnected} />
    );

    expect(actual).toMatchSnapshot();
  });

  it('should render a connected GID when isConnected param is not send', () => {
    const mockPositionX = 100;
    const mockPositionY = 100;

    const actual = shallow(<GID x={mockPositionX} y={mockPositionY} />);

    expect(actual).toMatchSnapshot();
  });
});
