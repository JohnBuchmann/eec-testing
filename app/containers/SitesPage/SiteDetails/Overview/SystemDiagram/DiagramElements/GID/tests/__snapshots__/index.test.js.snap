// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GID should render a connected GID when isConnected param is false 1`] = `
<Fragment>
  <desc>
    GID
  </desc>
  <text
    className="makeStyles-gidTitle-1"
    x={100}
    y={100}
  >
    GID
  </text>
  <Status
    height="24px"
    roundness="3px"
    showIcon={true}
    statusId={2}
    width="24px"
    x={97}
    y={107}
  />
</Fragment>
`;

exports[`GID should render a connected GID when isConnected param is not send 1`] = `
<Fragment>
  <desc>
    GID
  </desc>
  <text
    className="makeStyles-gidTitle-1"
    x={100}
    y={100}
  >
    GID
  </text>
  <Status
    height="24px"
    roundness="3px"
    showIcon={true}
    statusId={2}
    width="24px"
    x={97}
    y={107}
  />
</Fragment>
`;

exports[`GID should render a connected GID when isConnected param is true 1`] = `
<Fragment>
  <desc>
    GID
  </desc>
  <text
    className="makeStyles-gidTitle-1"
    x={100}
    y={100}
  >
    GID
  </text>
  <Status
    height="24px"
    roundness="3px"
    showIcon={false}
    statusId={5}
    width="24px"
    x={97}
    y={107}
  />
</Fragment>
`;

exports[`GID should render a connected GID with default props 1`] = `
<Fragment>
  <desc>
    GID
  </desc>
  <text
    className="makeStyles-gidTitle-1"
    x={0}
    y={0}
  >
    GID
  </text>
  <Status
    height="24px"
    roundness="3px"
    showIcon={true}
    statusId={2}
    width="24px"
    x={-3}
    y={7}
  />
</Fragment>
`;
